sql/core/src/main/scala/org/apache/spark/sql/sources/commands.scala

```diff
@@ -390,6 +406,7 @@ private[sql] class DefaultWriterContainer(
 
   override def commitTask(): Unit = {
     try {
+      assert(writer != null, "OutputWriter instance should have been initialized")
```

Comment:
Actually this line is just defensive. `writer` is initialized in `DefaultWriterContainer` rather than user code.